bin/dm_task_files.py

```diff
@@ -18,65 +18,83 @@
 import datman.utils
 import datman.dashboard as dashboard
 
-logging.basicConfig(level=logging.WARN,
-                    format="[%(name)s] %(levelname)s: %(message)s")
+logging.basicConfig(
+    level=logging.WARN, format="[%(name)s] %(levelname)s: %(message)s"
+)
 logger = logging.getLogger(os.path.basename(__file__))
 
 
 def main():
     args = docopt(__doc__)
-    study = args['<study>']
+    study = args["<study>"]
 
     config = datman.config.config(study=study)
-    subjects = datman.dashboard.get_study_subjects(study)
-    resources_dir = config.get_path('resources')
-    out_dir = config.get_path('task')
+    resources = get_resources_dirs(config)
+    out_dir = config.get_path("task")
     regex = get_regex(config)
 
     try:
         os.mkdir(out_dir)
     except OSError:
         pass
 
-    for subject in subjects:
+    for resource_folder in resources:
+        task_files = get_task_files(regex, resource_folder)
 
-        sessions = glob.glob(os.path.join(resources_dir, subject + '_*'))
-
-        if not sessions:
+        if not task_files:
             continue
 
-        for resource_folder in sessions:
-            task_files = get_task_files(regex, resource_folder)
-
-            if not task_files:
-                continue
-
-            session = os.path.basename(resource_folder)
-            dest_folder = os.path.join(out_dir, session)
-            try:
-                os.mkdir(dest_folder)
-            except OSError:
-                pass
+        session = os.path.basename(resource_folder)
+        dest_folder = os.path.join(out_dir, session)
+        try:
+            os.mkdir(dest_folder)
+        except OSError:
+            pass
```

Comment:
```suggestion
        os.makedirs(dest_folder, exist_ok=True)
```

Unless you want it to error out if intermediate directories don't exist!